library/Director/Import/Import.php

```diff
@@ -301,12 +301,20 @@ protected function storeRowset()
             }
 
             foreach ($newRows as $row) {
-                $db->insert('imported_row', $rows[$row]);
-                foreach ($this->rowProperties[$row] as $property) {
-                    $db->insert('imported_row_property', array(
-                        'row_checksum'      => $this->quoteBinary($row),
-                        'property_checksum' => $property
-                    ));
+                try {
+                    $db->insert('imported_row', $rows[$row]);
+                    foreach ($this->rowProperties[$row] as $property) {
+                        $db->insert('imported_row_property', array(
+                            'row_checksum'      => $this->quoteBinary($row),
+                            'property_checksum' => $property
+                        ));
+                    }
+                } catch (Exception $e) {
+                    throw new IcingaException(
+                        "Error while storing a row for '%s' into database: %s",
+                        $rows[$row]['object_name'],
```

Comment:
Do we always have an object_name here? If not, please handle that case too. The rest looks good!

Reply:
Yes we do:

https://github.com/Icinga/icingaweb2-module-director/blob/58c06890052a07db57ab8d3d26c481e08b3de4e8/library/Director/Import/Import.php#L262-L265

Reply:
Perfect, thanks!